File: task2.rb
# Task 2

a = ARGV[0].split(//)
b = ARGV[1].split(//)
s = -1;

if a.length == b.length
  a.each_index do |c|
    if a == b
      s = c
      break
    end #if
    a.push(a.shift)
  end #do
end #if

puts s > a.length - s ? a.length - s : s

File: task2.js
/* Task 2 */

var a = process.argv[2].split('');
var b = process.argv[3].split('');
var s = -1;

if (a.length === b.length) {
  for (let c = 0; c < a.length; c++) {
    if (a.every((x, i) => x === b[i])) {
      s = c;
      break;
    }
    a.push(a.shift());
  }
}

process.stdout.write((s > a.length - s ? a.length - s : s) + '\n');
